src/main/java/service/impl/FruitServiceImpl.java

```diff
@@ -0,0 +1,25 @@
+package service.impl;
+
+import java.util.Map;
+import model.FruitTransaction;
+import service.FruitService;
+import strategy.OperationHandler;
+
+public class FruitServiceImpl implements FruitService {
+    private final Map<FruitTransaction.Operation, OperationHandler> operationHandlers;
+
+    public FruitServiceImpl(Map<FruitTransaction.Operation, OperationHandler> operationHandlers) {
+        this.operationHandlers = operationHandlers;
+    }
+
+    @Override
+    public void applyTransaction(FruitTransaction transaction) {
+        OperationHandler handler = operationHandlers.get(transaction.getOperation());
+        if (handler != null) {
+            handler.apply(transaction);
+        } else {
+            throw new RuntimeException("No handler found for operation: "

```

Comment:
Throwing a raw RuntimeException is not a good practice. It's better to define a custom exception that is more descriptive of the problem domain.